### PR TITLE
Update packaging in AOTI path

### PR DESCRIPTION
**Pull request by angelayi**
Added a `aoti_package` path, dependent on https://github.com/pytorch/pytorch/pull/129895. Follow up will be to delete the `--output-dso-path`.

To export, use the `--output-aoti-package-path` to specify a file with a `.pt2` extension. This will generate an artifact containing all the AOTI generated files 
```bash
python3 torchchat.py export stories15M --output-aoti-package-path exportedModels/stories15M_artifacts_cpu.pt2 --device cpu
```

If we look into the contents of the package, we can see the following:
```
unzip -l exportedModels/stories15M_artifacts_cpu.pt2

Archive:  exportedModels/stories15M_artifacts_cpu.pt2
  Length      Date    Time    Name
---------  ---------- -----   ----
        1  09-13-2024 08:36   version
        3  09-13-2024 08:36   archive_format
   299452  09-13-2024 08:36   data/aotinductor/model/ca4wxt6cqzwws5meb44ouvurtpu5elwsqmsxayqx4cgmhvogmhys.cpp
       26  09-13-2024 08:35   data/aotinductor/model/ca4wxt6cqzwws5meb44ouvurtpu5elwsqmsxayqx4cgmhvogmhys_metadata.json
   644608  09-13-2024 08:36   data/aotinductor/model/ca4wxt6cqzwws5meb44ouvurtpu5elwsqmsxayqx4cgmhvogmhys.o
     1727  09-13-2024 08:36   data/aotinductor/model/ca4wxt6cqzwws5meb44ouvurtpu5elwsqmsxayqx4cgmhvogmhys_compile_flags.json
     1738  09-13-2024 08:36   data/aotinductor/model/ca4wxt6cqzwws5meb44ouvurtpu5elwsqmsxayqx4cgmhvogmhys_linker_flags.json
173405576  09-13-2024 08:36   data/aotinductor/model/ca4wxt6cqzwws5meb44ouvurtpu5elwsqmsxayqx4cgmhvogmhys.so
172941232  09-13-2024 08:36   data/aotinductor/model/cghfy3rmc5pj6omsgyzll5b7btki66najqkyd3544pmitg26ihvj.o
---------                     -------
347294363                     9 files
```

To run with Python:

```bash
python3 torchchat.py generate stories15M --aoti-package-path exportedModels/stories15M_artifacts_cpu.pt2 --prompt "Hello my name is"
```

To run with the C++ runner, first build the runner binary:
```bash
torchchat/utils/scripts/build_native.sh aoti
```

To run:
```bash
cmake-out/aoti_run exportedModels/stories15M_artifacts_cpu.pt2 -z `python3 torchchat.py where stories15M `/tokenizer.model -l 2 -i "Once upon a time"
```

AOTI changes in PyTorch also now allow users to not need to specify --device when generating, as we will save this information during export time as metadata and use it during runtime to determine what device to use. A followup can be to extend the metadata to save model information so that users do not need to specify tokenizer version to use during runtime.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +25,6 @@ system_info.txt
 
 # intermediate system file
 .DS_Store
+checkpoints/
+exportedModels/
+cmake-out/
```
[the remark this answers is not in the document]
Comment (angelayi) [reply .gitignore:30]: Here it is: https://github.com/pytorch/torchchat/pull/1143

**Comment by Jack-Khuu** [comment]
Weird that the pin bump in https://github.com/pytorch/torchchat/pull/1144 doesn't have the same errors as we see here....

Side question: Is it possible for us to push -l 2 into the dso as metadata as well or does that require looping through pytorch/pytorch? It's an arg when running the binary, but we know AOT what the arg should be

File: install/install_requirements.sh [hunk 2]
```diff
@@ -74,7 +74,7 @@ fi
 
 # pip packages needed by exir.
 REQUIREMENTS_TO_INSTALL=(
-  torch=="2.5.0.${PYTORCH_NIGHTLY_VERSION}"
+  torch=="2.6.0.${PYTORCH_NIGHTLY_VERSION}"
```
Comment (Jack-Khuu) [inline install/install_requirements.sh:77]: Just sanity checking that this was intentional?

**Comment by byjlw** [comment]
This command fails out of the box. I had to create the directory `exportedModels` for this to run
the command should create the directories in the specified path if they don't exist already.
`python3 torchchat.py export stories15M --output-aoti-package-path exportedModels/stories15M_artifacts_cpu.pt2 --device cpu`

**Comment by byjlw** [comment]
command to build the runner fails
`torchchat/utils/scripts/build_native.sh aoti`

**Comment by angelayi** [comment]
> the command should create the directories in the specified path if they don't exist already.

That should be fixed with https://github.com/pytorch/pytorch/pull/138919

**Comment by Jack-Khuu** [comment]
https://github.com/pytorch/torchchat/pull/1319/files

There's a bump here that's not actively being worked on, seems like it's not a free bump (+ the infra issue we're seeing here)

**Review by larryliu0820 — APPROVED** [review]

**Comment by byjlw** [comment]
currently running into this issue when trying to run using the runner.

```
(.venv) warden@Vikander:~/source/torchchat$ cmake-out/aoti_run exportedModels/exportedModels/stories15M_artifacts_cpu.pt2 -z `python3 torchchat.py where stories15M `/tokenizer.model -l 2 -i "Once upon a time"
2024-11-01:14:45:53,463 INFO     [sdpa_with_kv_cache.py:29] Loading custom ops library: /home/warden/source/torchchat/.venv/lib/python3.11/site-packages/executorch/extension/llm/custom_ops/libcustom_ops_aot_lib.so
terminate called after throwing an instance of 'std::runtime_error'
  what():  Failed to initialize zip archive: file open failed
Aborted (core dumped)

```

**Comment by byjlw** [comment]
> currently running into this issue when trying to run using the runner.
> 
> ```
> (.venv) warden@Vikander:~/source/torchchat$ cmake-out/aoti_run exportedModels/exportedModels/stories15M_artifacts_cpu.pt2 -z `python3 torchchat.py where stories15M `/tokenizer.model -l 2 -i "Once upon a time"
> 2024-11-01:14:45:53,463 INFO     [sdpa_with_kv_cache.py:29] Loading custom ops library: /home/warden/source/torchchat/.venv/lib/python3.11/site-packages/executorch/extension/llm/custom_ops/libcustom_ops_aot_lib.so
> terminate called after throwing an instance of 'std::runtime_error'
>   what():  Failed to initialize zip archive: file open failed
> Aborted (core dumped)
> ```

path issue. Just need to update the instructions in the first comment so it's 

`cmake-out/aoti_run exportedModels/stories15M_artifacts_cpu.pt2 -z `python3 torchchat.py where stories15M `/tokenizer.model -l 2 -i "Once upon a time"`
and not
`cmake-out/aoti_run exportedModels/exportedModels/stories15M_artifacts_cpu.pt2 -z `python3 torchchat.py where stories15M `/tokenizer.model -l 2 -i "Once upon a time"`

**Pull request merged**